src/auxiliary/auxiliary.jl

```diff
@@ -370,4 +370,14 @@ function download(src_url, file_path)
 
     return file_path
 end
+
+abstract type MeshFile{NDIMS} end
+
+struct GmshFile{NDIMS} <: MeshFile{NDIMS}
+    path::String
+end
+
+struct AbaqusFile{NDIMS} <: MeshFile{NDIMS}
+    path::String
+end
```

Comment:
Two questions:
* Should we call it `Abstract...` to be consistent with `AbstractEquations` etc.?
* Is this really intended as a wrapper/supertype for general mesh files? If not, maybe make it clear that this is only intended for _input_ mesh files from grid generators?

```suggestion
abstract type AbstractInputMeshFile{NDIMS} end

struct GmshFile{NDIMS} <: AbstractInputMeshFile{NDIMS}
    path::String
end

struct AbaqusFile{NDIMS} <: AbstractInputMeshFile{NDIMS}
    path::String
end
```

Reply:
Coming to think of it - why the abstract supertype at all? They don't seem to be used (yet), so following YAGNI, maybe a better approach would be - for now - to do this:
```suggestion
struct GmshFile{NDIMS}}
    path::String
end

struct AbaqusFile{NDIMS}
    path::String
end
```

Reply:
You have a point here. I'll simplify.